include/caffe/solver.hpp

```diff
@@ -218,6 +218,21 @@ class AdaDeltaSolver : public SGDSolver<Dtype> {
 };
 
```

Comment:
We need to cite the ADAM paper somewhere*.  I suggest putting a reference here, e.g. in a doxygen formatted comment like [this](https://github.com/BVLC/caffe/pull/2033/files#diff-2b4d88968d7da8d11c91d62d40aea3b9R152).  Eventually it would also be good to add sections to the [solver tutorial](http://caffe.berkeleyvision.org/tutorial/solver.html) on these new solvers, where the reference should also then be added.

*We probably also need to go back and add references for some of the other recently merged solvers.